### ISSUE TITLE
protoparse: trying to set extension field on a custom option whose type is message results in syntax error

### ISSUE THREAD

**Issue by bufdev** [jhump/protoreflect#307]
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/unittest_custom_options.proto#L281

```
src/google/protobuf/unittest_custom_options.proto:281:29:syntax error: unexpected '('
src/google/protobuf/unittest_custom_options.proto:282:29:syntax error: unexpected '('
src/google/protobuf/unittest_custom_options.proto:282:53:syntax error: unexpected ')'
```

It seems that the double parentheses are tripping up protoreflect

**Comment by jhump** [comment]
Welp, I _thought_ I had implemented support for this correctly. I see the intent to support this [in the grammar](https://github.com/jhump/protoreflect/blob/5374ad049d2606cb2c2a1284546b6f84f4c53a07/desc/protoparse/proto.y#L246).

But I think there is a legit shift-reduce conflict in the grammar; they are actually _many_ such conflicts reported by yacc, but I thought they were all harmless and that the generated parser would still work. In this case, I think the conflict causes the parser to not properly recognize this production. Luckily, I think it's a pretty easy fix.

**Comment by jhump** [comment]
Fixed in #316
